nottshcData/public_demographics_example.R

```diff
@@ -0,0 +1,258 @@
+# Postcode ----
+#' Postcode table
+#'
+#' @description Postcode information imported to SQL db from central public
+#' source.
+#' https://cdu-data-science-team.github.io/team-blog/posts/2021-05-14-index-of-multiple-deprivation/
+#' Links to \code{\link{tidy_postcodes}}
+#'
+#' @param select_vars Logical, specifying whether to include only selected
+#' variables (TRUE) or all variables from the database (FALSE).
+#'
+#' @param conn A database connection - stored in the database1 database but is
+#' generic to all data
+#' @param return String, specifying whether to return a "tbl_sql"
+#' connection or "tbl_df"
+#' @param local_area String, the postcode table covers the whole of England
+#' and has 2 million plus observations. Reducing the output to
+#' LocalAuthorityCode (511 and 512) or East Midlands makes this quicker to run.
+#'
+#' @section Last updated by:
+#' Zoë Turner
+#' @section Last updated date:
+#' 2022-01-21
+#'
+#' @export
+
+get_postcodes <- function(select_vars = TRUE,
+                          conn = conn_sql_server,
+                          local_area = c("local", "east_midlands", "all"),
+                          return = c("tbl_sql", "tbl_df")) {
+
+  # Check function arguments
+  return <- match.arg(return)
+  local_area <- match.arg(local_area)
+
+  # Create connection to server
+  if (!exists("conn_sql_database1")) {
+    assign("conn_sql_database1",
+      connect_sql(database = "database1"),
+      envir = globalenv()
+    )
+  }
+
+  # Create connection to table
+  db_data <- dplyr::tbl(conn, dbplyr::in_schema("schema", "PostCodes"))
+
+  if (local_area == "local") {
+    db_data <- db_data %>%
+      dplyr::filter(LocalAuthorityCode %in% c('511', '512'))
```

Comment:
Who picked these codes? is this "SQL" data?

Reply:
as LocalAuthorityCode is that a sql name?

Reply:
The LocalAuthorityCode column name and codes are from the public dataset 😃